src/ConfigResource.php

```diff
@@ -184,17 +157,26 @@ public function fetch($tree = false)
      * @param  array $config
      * @return array
      */
-    public function replaceKey($keys, $value, array $config)
+    public function replaceKey($keys, $value, array $config = null)
```

Comment:
Hm... BC Break, we can't set default value for a param, we will get warning/error. See: https://3v4l.org/imTjp

Reply:
This one is rather necessary else the whole config would need to be passed on the first call and this is a recursive function.  Passing the whole config is contrary to the class' job of managing a config and a file.  However as there are only a hand-ful of changes to replaceKey it would be possible to pass the config but I think that's not the best idea.